gc/base/standard/Scavenger.cpp

```diff
@@ -930,6 +930,69 @@ MM_Scavenger::calculateOptimumCopyScanCacheSize(MM_EnvironmentStandard *env)
     return cacheSize;
 }
 
+
+MMINLINE bool
+MM_Scavenger::activateSurvivorCopyScanCache(MM_EnvironmentStandard *env)
+{
+#if defined(OMR_GC_CONCURRENT_SCAVENGER)
+	MM_CopyScanCacheStandard *cache = (MM_CopyScanCacheStandard *)env->_inactiveSurvivorCopyScanCache;
+	if (NULL != cache) {
+		Assert_MM_true(NULL == env->_survivorCopyScanCache);
```

Comment:
I curious can it be racing here? There is not pair `env->_inactiveSurvivorCopyScanCache` and `env->_survivorCopyScanCache` atomic read. Is it possible that pointer would be flipped around between this two checks? Is something prevents this situation?

Reply:
Or I guess is this only place can switch inactive to active?

Reply:
You are absolutely right, I added this assert (and similar in tenure code) last moment, without testing it with the rest of the optimizatoin. Once the whole optimization is enabled there will be a race and another thread can set _survivorCopyScanCache to non-null between current thread reading _inactiveSurvivorCopyScanCache and setting it to null.

These type of assert are at a few other places too, but are really only safe in front of non-atomic store, when there is no race.

I'll remove them (even though it's benign atm, since _inactiveSurvivorCopyScanCache can't be non-null yet)

Reply:
I moved it down after successful atomic. For Tenure case, there was already one there.